spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/ReactiveSupportAutoConfiguration.java

```diff
@@ -51,6 +51,11 @@ public FluxToMessageChannelResultAdapter fluxToMessageChannelResultAdapter() {
 		return new FluxToMessageChannelResultAdapter();
 	}
 
+	@Bean
+	public StreamEmitterAnnotationBeanPostProcessor streamEmitterAnnotationBeanPostProcessor() {
```

Comment:
`BeanPostProcessor` s must be `static` bean methods: http://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-java-injecting-imported-beans:

> Also, be particularly careful with BeanPostProcessor and BeanFactoryPostProcessor definitions via @Bean. Those should usually be declared as static @Bean methods, not triggering the instantiation of their containing configuration class. Otherwise, @Autowired and @Value won’t work on the configuration class itself since it is being created as a bean instance too early.